### ISSUE TITLE
Staging Review finding: Required fields not marked as required

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#98226]
## Status
**2025-01-08] [Fran]** Jordan Wood (UX) asked about this in the platform design channel, and Matt Dingee advised us to add the required indicator to all fields in the DUW instead of starting an experimental design where we would have a statement on the introduction page. ([Slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1736187241045169)). Ticket to add the required statement to all question has been pulled into Sprint 21 (starts today).
**2024-12-23] [Fran]** Two tickets have been created for the Sitewide team - [UX](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20013) & [Frontend](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20167). The UX ticket is in Sprint 20 (last sprint of 2024) and the FE ticket is tagged for Sprint 21 (first sprint in the new year).

---

_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Public Websites
**Product:** Discharge Upgrade Wizard
**Feature:** 

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility), @erinrwhite (IA)

### Description

Throughout the form flow, every field is required to advance to the next step --- meaning, not providing a response results in an error message. But none of the inputs have the `required` prop on the component, which adds a visible "(*Required)" label and notifies users that they must provide a response for that field.

Note that the [design system includes this guidance](https://design.va.gov/components/form/):
> Properly indicate required form elements (the right thing will happen for you when you use the required property on form fields in the Design System).

There's a good case to be made for marking only optional fields and not marking required fields, as GOV.UK does. But it's a departure from the rest of VA.gov, and typically forms that follow that pattern include some sort of instruction at the beginning of the flow alerting users to expect all fields to be required.

<details><summary>Link, screenshot or steps to recreate</summary>

!["What was your branch of service?" question with no indication that it's required.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/80777-Brian-1.png?raw=true)

!["What was your branch of service?" question with an error message instructing the user ot choose a branch.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/80777-Brian-2.png?raw=true)

</details>

## Recommended action

If your intention is to test not marking required fields, then:
1. Include an instruction at the beginning of the form flow that tells users that all fields are required unless otherwise marked as optional.
2. I think this would benefit from going through the [experimental design process](https://design.va.gov/about/contributing-to-the-design-system/experimental-components-and-patterns) and getting buy-in from the Design System Council.

Otherwise, mark each field as required using the `required` prop on the component.

## References

- **Accessibility Defect Severity:** [3: Medium. Should be fixed in 1-3 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-3)
- **WCAG Success Criteria:** N/A
- **Modality:** N/A
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** User flow

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by FranECross** [comment]
Required component added to each question; released to prod and verified. Example screenshot below. Closing.

![Image](https://github.com/user-attachments/assets/b1a87248-7f6e-4674-8f4d-e2b004585abd)
